modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/APIEndpointCertificateUsageTestCase.java

```diff
@@ -0,0 +1,231 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.am.integration.clients.publisher.api.ApiException;
+import org.wso2.am.integration.clients.publisher.api.ApiResponse;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIMetadataDTO;
+import org.wso2.am.integration.clients.publisher.api.v1.dto.APIMetadataListDTO;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+/**
+ * This class contains integration tests for endpoint Certificate usage endpoint.
+ */
+public class APIEndpointCertificateUsageTestCase extends APIManagerLifecycleBaseTest {
+
+    private static final Log log = LogFactory.getLog(APIEndpointCertificateUsageTestCase.class);
+
+    private final String API_NAME_BASE = "APIEndpointCertificateUsageTestCase";
+    private final String API_CONTEXT_BASE = "APIEndpointCertificateUsageTestCase";
+    private final String API_VERSION_1_0_0 = "1.0.0";
+    private final String API_ENDPOINT_URL_BASE_1 = "https://abc.test1.com/resource";
+    private final String API_ENDPOINT_URL_BASE_2 = "https://abc.test2.com/resource";
+    private final String CERT_ALIAS = "cert00001";
+    private int apiCount;
+    private ArrayList<String> allAPIs;
+    private ArrayList<String> evenAPIs;
+    private String selectedEndpointUrl;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public APIEndpointCertificateUsageTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_USER},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        super.init(userMode);
+
+        this.evenAPIs = new ArrayList<>();
+        this.allAPIs = new ArrayList<>();
+
+        String providerName = user.getUserName();
+
+        Random random = new Random();
+        int limit = random.ints(20, 30).findFirst().getAsInt();
+
+        for(int i = 0; i < limit; i++) {
+            String apiName = API_NAME_BASE + "_" + i;
+            String apiContext = API_CONTEXT_BASE + "_" + i;
+            String apiEndPointUrl = API_ENDPOINT_URL_BASE_1 + i;
+
+            if(i % 2 == 0) {
+                apiEndPointUrl = API_ENDPOINT_URL_BASE_2 + i;
+                selectedEndpointUrl = apiEndPointUrl;
```

Comment:
`selectedEndpointUrl` is only initialized inside this condition. Is it the correct logic here?

Reply:
Yeah. This is the correct logic. We get the `apiEndPointUrl` corresponding to the last value divisible by 2 of the given range here as `selectedEndpointUrl`. There will always be a value for `selectedEndpointUrl`